fix:: Correct url encoding of environment name

OscarVanL opened the pull request:
Fixes https://github.com/PyGithub/PyGithub/issues/3262.

This changes the URL encoding function used for the GH Environment name from `quote` to `quote_plus`, which resolves `/` characters not being correctly URL-encoded in API calls.

I have updated the existing test cases with an environment name containing a `/`.

EnricoMi commented:
Thanks for fixing this. I think all places where we call `quote` should be fixed. Can you roll this fix out everywhere?

OscarVanL commented:
@EnricoMi 
I've fixed that up for other usages of `quote`. I skipped cases that related to commit SHAs, as they shouldn't contain any slashes anyway.

I didn't add any new test cases, except of course for the updated Environments API test I originally opened this fix for. I don't have time to setup the required integrations etc to do this. Hopefully the existing tests not failing is a good indication that the change doesn't cause any regressions.

Please double-check what API usages I've updated and if you're satisfied that's correct.

Prateikx commented on github/Repository.py:
```diff
@@ -1268,7 +1268,7 @@ def add_to_collaborators(self, collaborator: str | NamedUser, permission: Opt[st
         if isinstance(collaborator, github.NamedUser.NamedUser):
             collaborator = collaborator._identity
         else:
```
**Suggestion:** Consider using `urllib.parse.quote_plus` instead of `urllib.parse.quote` with an empty safe string.  `quote_plus` replaces spaces with plus signs, which is generally the expected behavior for URL encoding in query parameters and other parts of a URL, while `quote` with an empty safe string can lead to unexpected behavior with certain characters.
```suggestion
collaborator = urllib.parse.quote_plus(collaborator)
```

Prateikx commented on github/Repository.py:
```diff
@@ -1294,7 +1294,7 @@ def get_collaborator_permission(self, collaborator: str | NamedUser) -> str:
         if isinstance(collaborator, github.NamedUser.NamedUser):
             collaborator = collaborator._identity
         else:
```
**Suggestion:** Consider using `urllib.parse.quote_plus` instead of `urllib.parse.quote` with an empty safe string.  `quote_plus` replaces spaces with plus signs, which is generally the expected behavior for URL encoding in query parameters and other parts of a URL, while `quote` with an empty safe string can lead to unexpected behavior with certain characters.
```suggestion
collaborator = urllib.parse.quote_plus(collaborator)
```

Prateikx commented on github/Repository.py:
```diff
@@ -1932,7 +1932,7 @@ def create_secret(
         assert isinstance(unencrypted_value, str), unencrypted_value
         assert secret_type in ["actions", "dependabot"], "secret_type should be actions or dependabot"
```
**Suggestion:** Consider using `urllib.parse.quote_plus` instead of `urllib.parse.quote` with an empty safe string.  `quote_plus` replaces spaces with plus signs, which is generally the expected behavior for URL encoding in query parameters and other parts of a URL, while `quote` with an empty safe string can lead to unexpected behavior with certain characters.
```suggestion
secret_name = urllib.parse.quote_plus(secret_name)
```

Prateikx commented on github/Repository.py:
```diff
@@ -1980,7 +1980,7 @@ def get_secret(self, secret_name: str, secret_type: str = "actions") -> github.S
         assert isinstance(secret_name, str), secret_name
         assert secret_type in ["actions", "dependabot"], "secret_type should be actions or dependabot"
```
**Suggestion:** Consider using `urllib.parse.quote_plus` instead of `urllib.parse.quote` with an empty safe string.  `quote_plus` replaces spaces with plus signs, which is generally the expected behavior for URL encoding in query parameters and other parts of a URL, while `quote` with an empty safe string can lead to unexpected behavior with certain characters.
```suggestion
secret_name = urllib.parse.quote_plus(secret_name)
```

Prateikx commented on github/Repository.py:
```diff
@@ -2030,7 +2030,7 @@ def get_variable(self, variable_name: str) -> github.Variable.Variable:
         :rtype: github.Variable.Variable
         """
         assert isinstance(variable_name, str), variable_name
```
**Suggestion:** Consider using `urllib.parse.quote_plus` instead of `urllib.parse.quote` with an empty safe string.  `quote_plus` replaces spaces with plus signs, which is generally the expected behavior for URL encoding in query parameters and other parts of a URL, while `quote` with an empty safe string can lead to unexpected behavior with certain characters.
```suggestion
variable_name = urllib.parse.quote_plus(variable_name)
```

Prateikx commented on github/Repository.py:
```diff
@@ -2047,7 +2047,7 @@ def delete_secret(self, secret_name: str, secret_type: str = "actions") -> bool:
         """
         assert isinstance(secret_name, str), secret_name
         assert secret_type in ["actions", "dependabot"], "secret_type should be actions or dependabot"
```
**Suggestion:** Consider using `urllib.parse.quote_plus` instead of `urllib.parse.quote` with an empty safe string.  `quote_plus` replaces spaces with plus signs, which is generally the expected behavior for URL encoding in query parameters and other parts of a URL, while `quote` with an empty safe string can lead to unexpected behavior with certain characters.
```suggestion
secret_name = urllib.parse.quote_plus(secret_name)
```

Prateikx commented on github/Repository.py:
```diff
@@ -2058,7 +2058,7 @@ def delete_variable(self, variable_name: str) -> bool:
         :rtype: bool
         """
         assert isinstance(variable_name, str), variable_name
```
**Suggestion:** Consider using `urllib.parse.quote_plus` instead of `urllib.parse.quote` with an empty safe string.  `quote_plus` replaces spaces with plus signs, which is generally the expected behavior for URL encoding in query parameters and other parts of a URL, while `quote` with an empty safe string can lead to unexpected behavior with certain characters.
```suggestion
variable_name = urllib.parse.quote_plus(variable_name)
```

JLLeitschuh reviewed (approved):
LGTM!

JLLeitschuh commented:
It looks like some linting tasks are failing. I've rebased the branch. Can you pull the branch and fix the formatting issues? After that I'm happy to merge this.
